core/scheduler/service/executor_input_compiler.go

```diff
@@ -95,9 +123,19 @@ func (i InputCompiler) Compile(ctx context.Context, job *scheduler.Job, config s
 		return nil, err
 	}
 
-	jobAttributionLabels := fmt.Sprintf("project=%s,namespace=%s,job=%s", job.Tenant.ProjectName(), job.Tenant.NamespaceName(), job.Name)
-	if jobLables, ok := confs[JobAttributionLabelsKey]; ok {
-		confs[JobAttributionLabelsKey] = jobLables + "," + jobAttributionLabels
+	jobLabelsToAdd := map[string]string{
+		"project":   job.Tenant.ProjectName().String(),
+		"namespace": job.Tenant.NamespaceName().String(),
+		"job_name":  job.Name.String(),
+		"job_id":    job.ID.String(),
+	}
+	jobAttributionLabels := getJobLabelsString(jobLabelsToAdd)
+	if jobLabels, ok := confs[JobAttributionLabelsKey]; ok {
+		var separator string
+		if len(jobLabels) != 0 {
+			separator = ","
+		}
+		confs[JobAttributionLabelsKey] = jobLabels + separator + jobAttributionLabels
```

Comment:
please consider the above example I've provided to reduce the separator thing. this is to also reduce the number of required variable or constant (yes, I'm referring to separator where it's possible to be constant depending on the approach)

Reply:
considered